Userland/Libraries/LibWeb/Layout/ListItemMarkerBox.cpp

```diff
@@ -44,11 +44,40 @@ void ListItemMarkerBox::paint(PaintContext& context, PaintPhase phase)
 {
     if (phase != PaintPhase::Foreground)
         return;
-    Gfx::IntRect bullet_rect { 0, 0, 4, 4 };
-    bullet_rect.center_within(enclosing_int_rect(absolute_rect()));
+
     // FIXME: It would be nicer to not have to go via the parent here to get our inherited style.
     auto color = parent()->computed_values().color();
-    context.painter().fill_rect(bullet_rect, color);
+
+    auto enclosing = enclosing_int_rect(absolute_rect());
+    int marker_width = (int)enclosing.height() / 2;
+    Gfx::IntRect marker_rect { 0, 0, marker_width, marker_width };
+    marker_rect.center_within(enclosing);
+
+    switch (m_list_style_type) {
+    case CSS::ListStyleType::Square:
+        context.painter().fill_rect(marker_rect, color);
+        break;
+    case CSS::ListStyleType::Circle:
+        // For some reason for draw_ellipse() the ellipse is outside of the rect while for fill_ellipse() the ellipse is inside.
+        // Scale the marker_rect with sqrt(2) to get an ellipse arc (circle) that appears as if it was inside of the marker_rect.
+        marker_rect.set_height(marker_rect.height() / 1.41);
+        marker_rect.set_width(marker_rect.width() / 1.41);
+        marker_rect.center_within(enclosing);
+        context.painter().draw_ellipse_intersecting(marker_rect, color);
+        break;
+    case CSS::ListStyleType::Decimal:
+        context.painter().draw_text(enclosing, String::formatted("{}.", m_index), Gfx::TextAlignment::Center);
+        break;
+    case CSS::ListStyleType::Disc:
+        context.painter().fill_ellipse(marker_rect, color);
+        break;
+    case CSS::ListStyleType::None:
+        return;
+        break;
```

Comment:
No need for both `return` and `break` :)

Reply:
Good point ;)